marshal/claim.go

```diff
@@ -261,6 +263,13 @@ func (c *claim) teardown(releasePartition bool) bool {
 	// is reasonable. Held because of using offsetCurrent below.
 	c.lock.RLock()
 	defer c.lock.RUnlock()
+
+	// Advise the consumer that this claim is terminating, this is so that the consumer
+	// can release other claims if we've lost part of a topic
+	if c.consumer != nil {
+		go c.consumer.claimTerminated(c, releasePartition)
```

Comment:
I know you guys like to use go routines to do stuff outside locks, but this can be bad sometimes. For example, in this case, if we call Terminate, which ends up calling this function, now, it's unclear when stuff actually get terminated because things are happening async...and will need to wait for additional things if I want to cleanup gracefully.